### PR TITLE
SONARCLOUD - Sprint 7

### PR DESCRIPTION
**Pull request by cindyslittleplanet**
This PR aims to fix all issues brought up by SonarCloud. 

You can access the board [here](https://sonarcloud.io/dashboard?id=RGPosadas_Mull).

Some of the issues fix include: 
- Resolving all code smells
- Increasing the code coverage
- Excluding `user.mockdata.ts` since it kept bringing up false security hotspots and false code duplication

### CODE REVIEW

File: apps/mull-api/src/app/event/event.resolver.ts [hunk 1]
```diff
@@ -86,6 +86,6 @@ export class EventResolver {
 
   @ResolveField(/* istanbul ignore next */ () => Media)
   async image(@Parent() event: Event) {
-    return await this.eventService.getEventImage(event.id);
+    return this.eventService.getEventImage(event.id);
```
Comment (r-saba) [inline apps/mull-api/src/app/event/event.resolver.ts:89]: I wasn't aware that `async` wraps returns with a promise for us and we don't have to explicitly return a promise. Good to know!

File: apps/mull-api/src/app/user/user.resolver.ts [hunk 2]
```diff
@@ -39,29 +39,38 @@ export class UserResolver {
...
-    @Args('newAvatar', { type: /* istanbul ignore next */ () => GraphQLUpload, nullable: true })
-    newAvatar: FileUpload
-  ) {
-    const user = await this.userService.getUser(userInput.id);
+  private async handleNewAvatar(
+    newAvatar: FileUpload,
+    user: User,
+    userInput: UpdateUserInput
+  ): Promise<UpdateUserInput> {
+    var media: Media;
```
Comment (r-saba) [inline apps/mull-api/src/app/user/user.resolver.ts]:
```suggestion
    let media: Media;
```

File: apps/mull-api/src/app/user/user.resolver.ts [hunk 3]
```diff
@@ -39,29 +39,38 @@ export class UserResolver {
...
     if (newAvatar) {
       if (user.avatar) {
         var oldMediaId = user.avatar.id;
```
Comment (r-saba) [inline apps/mull-api/src/app/user/user.resolver.ts]:
```suggestion
        const oldMediaId = user.avatar.id;
```

**Review by r-saba — APPROVED** [review]
LGTM! Left 2 changes that should be fixed but no re-request necessary.

**Review by RGPosadas — APPROVED** [review]
Thanks for fixing all this!

**Pull request merged**